wandb/sdk/artifacts/artifact.py

```diff
@@ -1686,6 +1686,78 @@ def download(
         root = root or self._default_root()
         self._add_download_root(root)
 
+        if wandb.run is None:
+            with wandb.init(
+                entity=self._source_entity,
+                project=self._source_project,
+                job_type="auto",
+                settings=wandb.Settings(silent="true"),
+            ):
+                return self._run_artifact_download(
+                    root=root,
+                    recursive=recursive,
+                    allow_missing_references=allow_missing_references,
+                )
+        else:
+            return self._run_artifact_download(
+                root=root,
+                recursive=recursive,
+                allow_missing_references=allow_missing_references,
+            )
+
+    def _run_artifact_download(
+        self,
+        root: str,
+        recursive: bool = False,
+        allow_missing_references: bool = False,
+    ) -> FilePathStr:
+        assert wandb.run is not None, "failed to initialize run"
+        run = wandb.run
+        if run._settings._require_nexus:
+            if not run._backend or not run._backend.interface:
+                raise NotImplementedError
+            if run._settings._offline:
+                raise NotImplementedError("cannot download in offline mode")
+            # Start the download process in the user process too, to handle reference downloads
+            self._download(
+                root=root,
+                recursive=recursive,
+                allow_missing_references=allow_missing_references,
+            )
+            handle = run._backend.interface.deliver_download_artifact(
+                self.qualified_name,
+                root,
+                recursive,
+                allow_missing_references,
+            )
+            result = handle.wait(timeout=-1)
+            if result is None:
+                handle.abandon()
+            assert result is not None
+            response = result.response.download_artifact_response
+            if response.error_message:
+                raise ValueError(
+                    f"Error downloading artifact: {response.error_message}"
+                )
+            download_path = response.file_download_path
+            return FilePathStr(download_path)
+        return self._download(
+            root=root,
+            recursive=recursive,
+            allow_missing_references=allow_missing_references,
+        )
+
+    def _download(
+        self,
+        root: str,
+        recursive: bool = False,
+        allow_missing_references: bool = False,
+    ) -> FilePathStr:
+        # todo: remove once artifact reference downloads are supported in nexus
+        require_nexus = False
+        if wandb.run is not None:
```

Comment:
Nit: this is always true, remove the if. If mypy complains that `wandb.run` can be `None`, do `assert wandb.run is not None`. This will make the code easier to understand.